pkgs/mcaptcha/default.nix

```diff
@@ -0,0 +1,190 @@
+{
+  mkYarnPackage,
+  fetchYarnDeps,
+  fetchFromGitHub,
+  rustPlatform,
+  lib,
+  openssl,
+  pkg-config,
+  perl,
+  nixosTests,
+}: let
+  rootSrc = fetchFromGitHub {
+    owner = "mCaptcha";
+    repo = "mCaptcha";
+    rev = "f337ee0643d88723776e1de4e5588dfdb6c0c574";
+    sha256 = "sha256-UP7V2TfbW+KJpNAQLxQIcRsT9ZWYGVkS13XxMbHEH2I=";
+  };
+
+  releaseDate = "2023-07-04";
+
+  frontend = let
+    src = rootSrc;
+  in
+    mkYarnPackage {
+      pname = "mcaptcha-frontend";
+      version = "unstable-${releaseDate}";
+      inherit src;
+
+      packageJSON = ./frontend-package.json;
+      offlineCache = fetchYarnDeps {
+        yarnLock = "${src}/yarn.lock";
+        sha256 = "sha256-GyWjQdFJ+hEuR4PebhYzFwiuyMyamRY5GPaJ7rK9Rsc=";
+      };
+
+      buildPhase = ''
+        export HOME=$(mktemp -d)
+        yarn --offline build
+      '';
+
+      doCheck = true;
+      checkPhase = ''
+        CI=true yarn test
+      '';
+
+      # Copied from `make frontend` in the mCaptcha Makefile:
+      # https://github.com/mCaptcha/mCaptcha/blob/f337ee0643d88723776e1de4e5588dfdb6c0c574/Makefile#L130-L147
```

Comment:
notes from chatting with @realaravinth: would prefer for us to use `make frontend` here, and is open to changing the Makefile to make this nix compatible

Reply:
We're going to skip this for now. We'll revisit this if this ever breaks in the future.